src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -95,15 +95,23 @@ public static void SetCurrentCircuitHost(CircuitHost circuitHost)
 
         public IDispatcher Dispatcher { get; }
 
-        public Task<IEnumerable<string>> PrerenderComponentAsync(Type componentType, ParameterCollection parameters)
+        public Task<ComponentRenderedText> PrerenderComponentAsync(Type componentType, ParameterCollection parameters)
         {
             return Dispatcher.InvokeAsync(async () =>
             {
                 var result = await Renderer.RenderComponentAsync(componentType, parameters);
+
+                await OnCircuitOpenedAsync(CancellationToken.None);
```

Comment:
I think this is worth a comment that this is done here because the circuit is considered *open* but not connected at this point during prerednering.

Reply:
Added a comment